src/python/pants/engine/internals/dep_rules_test.py

```diff
@@ -0,0 +1,99 @@
+# Copyright 2022 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import annotations
+
+import logging
+from collections import namedtuple
+
+import pytest
+
+from pants.backend.experimental.visibility.register import BuildFileVisibilityRules
```

Comment:
borrowing an implementation in order to test the `extend` feature.

Reply:
agree, my mental capacity was at an end. Can make a dummy implementation for the tests.

Reply:
Thanks!